apps/backend/src/handlers/stream-handlers.ts

```diff
@@ -160,14 +208,27 @@ export function createStreamHandlers({ streamService, workspaceService }: Depend
 
     async mute(req: Request, res: Response) {
       const userId = req.userId!
+      const workspaceId = req.workspaceId!
       const { streamId } = req.params
-      const { muted } = req.body
 
-      if (typeof muted !== "boolean") {
-        return res.status(400).json({ error: "muted must be a boolean" })
+      const result = muteSchema.safeParse(req.body)
+      if (!result.success) {
+        return res.status(400).json({
+          error: "Validation failed",
+          details: result.error.flatten().fieldErrors,
+        })
+      }
+
+      try {
+        await streamService.validateStreamAccess(streamId, workspaceId, userId)
+      } catch (error) {
+        if (error instanceof StreamNotFoundError) {
```

Comment:
I wonder if this could be centralized and moved into a reusable error handler? So that when this error is thrown, we **always** return the same error no matter which handler it is?